src/ws/cockpitcertificate.h

```diff
@@ -25,8 +25,7 @@
 
 G_BEGIN_DECLS
 
-gchar *             cockpit_certificate_locate   (gboolean create_if_necessary,
-                                                  GError **error);
+gchar *             cockpit_certificate_create_selfsigned  (GError **error);
```

Comment:
Keep the old function as a wrapper around the new one, doing the `GError` conversion and self-signed creation if necessary.  Otherwise you need to do this dance three separate times below.

Reply:
I really don't like the old function. Location and creation are two different things, so I'd rather keep the split. I can create a GError wrapper for cockpit_certificate_locate(), though.

Reply:
Ah, probably you meant it like that. I added a cockpit_certificate_locate_gerror() now.